### PR TITLE
H3: ignore any number of unknown settings

### PR DESCRIPTION
**Pull request by stammw**
Addressing a bug that emerged from interop test against gQuic.

### CODE REVIEW

File: quinn-h3/src/proto/settings.rs [hunk 1]
```diff
@@ -19,6 +19,15 @@ macro_rules! setting_identifiers {
 
 impl SettingId {
     const NONE: SettingId = SettingId(0);
+
+    fn is_supported(id: SettingId) -> bool {
```
Comment (Ralith) [inline quinn-h3/src/proto/settings.rs:23]:
```suggestion
    fn is_supported(self) -> bool {
```

This would be a bit more convenient to call.

**Comment by Ralith** [comment]
CI failures look unrelated, due to a deprecation in an upstream crate.

**Comment by stammw** [comment]
Yeah, there is a breaking change in pin_project... I'm on it, I'll get this right soon.

**Comment by stammw** [comment]
It should be fine now.

**Pull request merged**